.azure-pipelines/client.yml

```diff
@@ -58,57 +58,21 @@ jobs:
       displayName: 'Component Detection'
 
     - task: Maven@3
-      displayName: 'Generate project site'
+      displayName: 'Generate Maven project site, including JavaDocs, SpotBugs, and CheckStyle reports'
       inputs:
         mavenPomFile: $(pomFile)
-        options: '--batch-mode'
+        options: '--batch-mode -Pextra-modules -DskipTests -Dgpg.skip'
         mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
         javaHomeOption: 'JDKVersion'
         jdkVersionOption: '1.11'
         jdkArchitectureOption: 'x64'
         publishJUnitResults: false
-        goals: 'site'
-
-    - task: Maven@3
-      displayName: 'Install packages to local cache and generate SpotBugs report'
-      inputs:
-        mavenPomFile: $(pomFile)
-        options: '--batch-mode -Pspotbugs-reporting -DskipTests -Dgpg.skip'
-        mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
-        javaHomeOption: 'JDKVersion'
-        jdkVersionOption: '1.11'
-        jdkArchitectureOption: 'x64'
-        publishJUnitResults: false
-        goals: 'install'
-
-    - task: Maven@3
-      displayName: 'Generate Checkstyle report'
-      inputs:
-        mavenPomFile: $(pomFile)
-        options: '--batch-mode'
-        mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
-        javaHomeOption: 'JDKVersion'
-        jdkVersionOption: '1.11'
-        jdkArchitectureOption: 'x64'
-        publishJUnitResults: false
-        goals: 'checkstyle:checkstyle-aggregate'
-
-    - task: Maven@3
-      displayName: 'Aggregate Javadocs'
-      inputs:
-        mavenPomFile: $(pomFile)
-        options: '--batch-mode'
-        mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
-        javaHomeOption: 'JDKVersion'
-        jdkVersionOption: '1.11'
-        jdkArchitectureOption: 'x64'
-        publishJUnitResults: false
-        goals: 'javadoc:aggregate'
+        goals: 'install site:site site:stage'
```

Comment:
Does it make sense to explicitly list out the stylecheck and spotbug goals so it is clear that they are being called?

Reply:
I would probably not do this - `mvn site:site site:stage` will be documented elsewhere as the way to generate all reports.

Reply:
For a non-java person it certainly isn't clear to me that site is causing those goals to run. Even looking at the docs at https://maven.apache.org/plugins/maven-site-plugin/site-mojo.html it isn't obvious to me that it is trigging those goals. Can you point me at what you look at to figure out the goal chain?

Reply:
Maven site is configured in the `reporting` section of the maven pom, where readers can see things like checkstyle, spotbugs, checkstyle, etc are configured.